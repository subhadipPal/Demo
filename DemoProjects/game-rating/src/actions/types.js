export const FETCH_GAME_LIST = "FETCH_GAME_LIST";
export const SET_GAME_RATING = "SET_GAME_RATING";
export const SORT_GAMES_BY_RATING = "SORT_GAMES_BY_RATING";
export const RANDOMLY_RATE = "RANDOMLY_RATE";
