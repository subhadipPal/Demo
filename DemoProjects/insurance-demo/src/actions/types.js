export const DISPLAY_MODULES = "DISPLAY_MODULES";
export const ADD_ITEM_TO_CART = "ADD_ITEM_TO_CART";
export const DELETE_ITEM_FROM_CART = "DELETE_ITEM_FROM_CART";
export const DISPLAY_CART_CONTENTS = "DISPLAY_CART_CONTENTS";
