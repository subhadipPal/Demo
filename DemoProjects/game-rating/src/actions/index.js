import { FETCH_GAME_LIST, SET_GAME_RATING, SORT_GAMES_BY_RATING, RANDOMLY_RATE } from "./types";
import axios from "axios";

export const fetchGameList = () => async dispatch => {
  var response = await axios.get("data/data.json");
  dispatch({
    type: FETCH_GAME_LIST,
    payload: response.data
  });
};

export const setGameRating = (newRating, oldRating) => {
  return {
    type: SET_GAME_RATING,
    payload: { newRating, oldRating }
  };
};

export const sortGamesByRating = sortDirection => {
  return {
    type: SORT_GAMES_BY_RATING,
    payload: sortDirection
  };
};

export const randomlyRateGames = () => {
  return {
    type: RANDOMLY_RATE
  };
};
