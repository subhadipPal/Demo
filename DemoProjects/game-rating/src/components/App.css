body {
  margin: 0px;
}

.App {
  min-height: 100vh;
  background-color: #f1f1f1;
  background-size: cover;
  font-family: "Times New Roman", Times, serif;
}
.App-header {
  background-color: #282c34;
  min-height: 50px;
  display: flex;
  flex-direction: column;
  justify-content: center;
}
.App-header-text {
  font-size: 24px;
  color: white;
  margin-left: 15px;
}
.App-content {
  margin: 10px;
}
.games-grid {
  display: grid;
  grid-gap: 50px;
  grid-template-columns: auto auto auto auto;
  margin-top: 20px;
  width: 100%;
}
.grid-item {
  background-color: #fff;
  border: 1px solid #555;
  padding: 20px;
  text-align: center;
}
.game-title {
  font-size: 20px;
  font-style: italic;
}
.game-rating {
  border-top: 2px dashed #f1f1f1;
  margin-top: 10px;
  padding: 10px;
}
.field-dropdown {
  margin: 10px 0px 0px 5px;
  color: #00b1c7;
  font-size: 14px;
  outline: 0;
  border-width: 0 0 2px;
}
.dropdown-label {
  color: crimson;
}

.header-button {
  margin: 5px;
  cursor: pointer;
  border: none;
  color: #fff;
  font-size: 14px;
  text-align: center;
  padding: 10px;
  background: #555 none;
  border-radius: 5px;
}
.header-button:hover {
  background: #ccc none;
  color: #555;
}
