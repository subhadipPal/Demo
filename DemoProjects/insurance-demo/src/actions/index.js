import {
  DISPLAY_MODULES,
  ADD_ITEM_TO_CART,
  DELETE_ITEM_FROM_CART,
  DISPLAY_CART_CONTENTS
} from "./types";

import history from "../history";

export const displayModules = () => {
  return {
    type: DISPLAY_MODULES,
    payload: [
      {
        id: 0,
        icon: "bicycle",
        name: "Bike",
        coverageMin: 0,
        coverageMax: 3000,
        risk: 30
      },
      {
        id: 1,
        icon: "gem",
        name: "Jewelry",
        coverageMin: 500,
        coverageMax: 10000,
        risk: 5
      },
      {
        id: 2,
        icon: "microchip",
        name: "Electronics",
        coverageMin: 500,
        coverageMax: 6000,
        risk: 35
      },
      {
        id: 3,
        icon: "football ball",
        name: "Sports Equipment",
        coverageMin: 0,
        coverageMax: 20000,
        risk: 30
      }
    ]
  };
};

export const addItemToCart = module => {
  return {
    type: ADD_ITEM_TO_CART,
    payload: module
  };
};

export const deleteItemFromCart = (timeStamp, price) => {
  return {
    type: DELETE_ITEM_FROM_CART,
    payload: { timeStamp, price }
  };
};

export const displayCartContents = () => {
  history.push("/cart");
  return {
    type: DISPLAY_CART_CONTENTS
  };
};
