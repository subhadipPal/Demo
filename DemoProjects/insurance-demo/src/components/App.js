import React, { useEffect } from "react";
import { useDispatch } from "react-redux";
import { Router, Route, Switch, Redirect } from "react-router-dom";

import history from "../history";
import LandingPage from "./home/LandingPage";
import { displayModules } from "../actions";
import Cart from "./home/Cart";

const App = () => {
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(displayModules());
  }, [dispatch]);

  return (
    <Router history={history}>
      <Switch>
        <Route path="/" exact component={LandingPage}></Route>
        <Route path="/cart" exact component={Cart}></Route>
        <Route render={() => <Redirect to="/" />} />
      </Switch>
    </Router>
  );
};

export default App;
