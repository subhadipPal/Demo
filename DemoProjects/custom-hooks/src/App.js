import React from 'react';
import './App.css';
import { Random } from './components/Random_V1';
import { RandomV2 } from './components/Random_V2';
import { Tag } from './components/Tag_V1';
import { TagV2 } from './components/Tag_V2';

export const App = () => (
  <div>
    <h1>Random Gif App</h1>
    <div className="main-container">
      <Random />
      <Tag />
      <RandomV2 />
      <TagV2 />
    </div>
  </div>
);
