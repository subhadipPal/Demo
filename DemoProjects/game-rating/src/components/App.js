import React from "react";
import { connect } from "react-redux";
import { Observable } from "rxjs/Rx";
import { range, of } from "rxjs";
import { concatMap, delay } from "rxjs/operators";

import { sortGamesByRating, randomlyRateGames } from "../actions";
import GameList from "./GameList";
import "./App.css";

class App extends React.PureComponent {
  constructor(props) {
    super(props);
    this.sortButtonAscRef = React.createRef();
    this.sortButtonDescRef = React.createRef();
  }
  componentDidMount() {
    const { sortGamesByRating, randomlyRateGames } = this.props;
    Observable.fromEvent(this.sortButtonAscRef.current, "click").subscribe(() => {
      sortGamesByRating("ASC");
    });
    Observable.fromEvent(this.sortButtonDescRef.current, "click").subscribe(() => {
      sortGamesByRating("DESC");
    });

    range(1, 10)
      .pipe(concatMap(i => of(i).pipe(delay(1000 + Math.random() * 4000))))
      .subscribe(() => {
        randomlyRateGames();
      });
  }
  render() {
    return (
      <div className="App">
        <div className="App-header">
          <div className="App-header-text">Game Ratings</div>
        </div>
        <div className="App-content">
          <div className="list-buttons-container">
            <button className="header-button" ref={this.sortButtonAscRef} name="sortAscButton">
              Sort by Ratings (Ascending)
            </button>
            <button className="header-button" ref={this.sortButtonDescRef} name="sortDescButton">
              Sort by Ratings (Decending)
            </button>
          </div>
          <GameList />
        </div>
      </div>
    );
  }
}
const mapStateToProps = state => {
  return {
    games: state.games
  };
};
export default connect(mapStateToProps, { sortGamesByRating, randomlyRateGames })(App);
