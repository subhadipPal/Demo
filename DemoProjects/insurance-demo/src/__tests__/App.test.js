import React from "react";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";

import App from "../components/App";
import history from "../history";
import { displayModules } from "../actions";
import { DISPLAY_MODULES } from "../actions/types";

const mockStore = configureStore([]);

describe("App Component test", () => {
  let store = {};
  let wrappedComponent = {};
  const expectedActions = {
    type: DISPLAY_MODULES,
    payload: [
      {
        id: 0,
        icon: "bicycle",
        name: "Bike",
        coverageMin: 0,
        coverageMax: 3000,
        risk: 30
      },
      {
        id: 1,
        icon: "gem",
        name: "Jewelry",
        coverageMin: 500,
        coverageMax: 10000,
        risk: 5
      },
      {
        id: 2,
        icon: "microchip",
        name: "Electronics",
        coverageMin: 500,
        coverageMax: 6000,
        risk: 35
      },
      {
        id: 3,
        icon: "football ball",
        name: "Sports Equipment",
        coverageMin: 0,
        coverageMax: 20000,
        risk: 30
      }
    ]
  };
  beforeEach(() => {
    store = mockStore({
      data: {
        modules: [],
        cart: [],
        total: 0
      }
    });
    store.dispatch = jest.fn(displayModules);
    wrappedComponent = renderer.create(
      <Provider store={store}>
        <App />
      </Provider>
    );
  });

  it("should render with given state from Redux store", () => {
    expect(wrappedComponent.toJSON()).toMatchSnapshot();
  });
  it("should have an app from Landing Page", () => {
    expect(wrappedComponent.root.findByProps({ className: "app" })).toBeDefined();
  });

  it("should show landing page for default route", () => {
    expect(wrappedComponent.root.findByProps({ className: "shopping cart icon" })).toBeDefined();
  });
  it("should show cart page for /cart route", () => {
    history.push("/cart");
    expect(wrappedComponent.root.findByProps({ className: "backward icon" })).toBeDefined();
  });
  it("should redirect to landing page for unmatched 404 routes", () => {
    history.push("/someRandomRoute");
    expect(wrappedComponent.root.findByProps({ className: "shopping cart icon" })).toBeDefined();
  });
  it("should dispatch displayModules action on app mount", async () => {
    const actualAction = await store.dispatch();
    expect(actualAction).toEqual(expectedActions);
  });
});
