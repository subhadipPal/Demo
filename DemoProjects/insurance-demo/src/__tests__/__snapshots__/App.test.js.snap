// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component test should render with given state from Redux store 1`] = `
<div
  className="app"
>
  <div
    className="header"
  >
    <div
      className="text"
    >
      Insurance modules
    </div>
    <i
      className="shopping cart icon"
      onClick={[Function]}
    >
      <span
        className="badge"
      >
        0
      </span>
    </i>
  </div>
  <div
    className="body"
  />
</div>
`;
